addons/account/models/account_move.py

```diff
@@ -134,6 +134,8 @@ def write(self, vals):
             self.assert_balanced()
         else:
             res = super(AccountMove, self).write(vals)
+        if 'analytic_account_id' in vals or 'analytic_tag_ids' in vals:
```

Comment:
1) We don't want analytic entries are created again at each write.
2) You don't unlink previous analytic entries.
3) analytic_account_id / analytic_tag_ids are fields on account.move.line, not account.move.

The current behavior is just fine: creating the analytic entries when posting the account.move. I don't understand the goal of your PR since you shouldn't be able to edit journal items when the journal entry is posted.